driver-core/src/main/java/com/datastax/driver/core/querybuilder/Utils.java

```diff
@@ -33,7 +33,7 @@
 abstract class Utils {
 
     private static final Pattern alphanumeric = Pattern.compile("\\w+"); // this includes _
-    private static final Pattern cnamePattern = Pattern.compile("\\w+(?:\\[.+\\])?");
+    private static final Pattern cnamePattern = Pattern.compile("\\w+(?:\\[.+\\]|\\.\\w+)?");
```

Comment:
This does not handle "a.b.c" if there are multiple levels of nesting (is it possible?)